src/frequenz/sdk/_internal/__init__.py

```diff
@@ -3,7 +3,8 @@
 
 """Utility types and functions for internal use."""
 
+from .asyncio import cancel_and_await
 from .singleton_meta import SingletonMeta
 
 # Explicitly declare the public API.
-__all__ = ["SingletonMeta"]
+__all__ = ["SingletonMeta", "cancel_and_await"]
```

Comment:
This module has generic name `_internal`. 

Because of that i would say that separation into submodule `asyncio` is fine.

Reply:
it is in a submodule already.  are you saying we shouldn't export reexport the function from `_internal`?

Reply:
I've updated the PR with that assumption.

Reply:
Yes that what I meant :)